pkg/operator/operator.go

```diff
@@ -88,8 +89,11 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 	azClient, err := instance.CreateAZClient(ctx, azConfig)
 	lo.Must0(err, "creating Azure client")
 
-	vnetGUID, err := getVnetGUID(azConfig, options.FromContext(ctx).SubnetID)
-	lo.Must0(err, "getting VNET GUID")
+	if options.FromContext(ctx).VnetGUID == "" && options.FromContext(ctx).NetworkPluginMode == consts.NetworkPluginModeOverlay {
```

Comment:
IMO leave a comment here for why we're supporting both ways for now, and what the future holds here.

Is the idea that: If it's passed via Helm (which AKS RP will do, but standalone Karpenter users probably will not), use the value from Helm and don't make the call to the networking RP?